tnt/src/main/java/org/dcsa/conformance/standards/tnt/TntScenarioListBuilder.java

```diff
@@ -114,7 +130,21 @@ private static TntScenarioListBuilder getEvents() {
             new TntGetEventsAction(
                 subscriberPartyName,
                 publisherPartyName,
-                previousAction,
+              (TntAction) previousAction,
                 componentFactory.getEventSchemaValidators()));
   }
+
+  private static TntScenarioListBuilder getEventsBadRequest() {
+    TntComponentFactory componentFactory = threadLocalComponentFactory.get();
```

Comment:
Just trying to understand, why do we need the bad request as a separate action?

Reply:
Can't we accomplish this by passing the GetEvents action with isBadRequest? Just checking as it appeared as a scenario than an action to me. It's totally fine with me if the workflow demands it to be like this. You can ignore this comment.

Reply:
I even thought about it but too many booleans travelling across the classes makes code bit unreadable. Hence I used this option also the same in booking as well.